Broken with SublimeLinter4

With sublimelinter4, this breaks

The temp files get parsed as SASS instead of SCSS, leading to the dreaded "1:1  error  Please check validity of the block starting from line #1  Fatal"

But if I run the same command on the command line against the real file (instead of the temp in /var) , I get expected linting output:
`/Users/amclin/Documents/Clients/VFc/cms/vfcorp-ui/src/design/node_modules/.bin/sass-lint --verbose --no-exit --format stylish /path/to/_dynamicmediabanner.scss`

Here's the log

```
startup, version: 3143 osx x64 channel: stable
executable: /Applications/Sublime Text.app/Contents/MacOS/Sublime Text
working dir: /
packages path: /Users/amclin/Library/Application Support/Sublime Text 3/Packages
state path: /Users/amclin/Library/Application Support/Sublime Text 3/Local
zip path: /Applications/Sublime Text.app/Contents/MacOS/Packages
zip path: /Users/amclin/Library/Application Support/Sublime Text 3/Installed Packages
ignored_packages: ["Vintage"]
pre session restore time: 0.285977
using gpu buffer for window
using gpu buffer for window
startup time: 0.414563
no such context .string-single in Packages/SCSS/SCSS.tmLanguage
no such context .string-double in Packages/SCSS/SCSS.tmLanguage
no such context .media-query-list in Packages/SCSS/SCSS.tmLanguage
no such context .string-double in Packages/SCSS/SCSS.tmLanguage
no such context .string-single in Packages/SCSS/SCSS.tmLanguage
no such context .media-query-list in Packages/SCSS/SCSS.tmLanguage
reloading plugin Default.auto_indent_tag
reloading plugin Default.block
reloading plugin Default.comment
reloading plugin Default.convert_syntax
reloading plugin Default.copy_path
reloading plugin Default.delete_word
reloading plugin Default.detect_indentation
reloading plugin Default.duplicate_line
reloading plugin Default.echo
reloading plugin Default.exec
reloading plugin Default.fold
reloading plugin Default.font
reloading plugin Default.goto_line
reloading plugin Default.history_list
reloading plugin Default.indentation
reloading plugin Default.install_package_control
reloading plugin Default.kill_ring
reloading plugin Default.mark
reloading plugin Default.new_templates
reloading plugin Default.open_context_url
reloading plugin Default.open_in_browser
reloading plugin Default.pane
reloading plugin Default.paragraph
reloading plugin Default.paste_from_history
reloading plugin Default.profile
reloading plugin Default.quick_panel
reloading plugin Default.run_syntax_tests
reloading plugin Default.save_on_focus_lost
reloading plugin Default.scroll
reloading plugin Default.set_unsaved_view_name
reloading plugin Default.settings
reloading plugin Default.show_scope_name
reloading plugin Default.side_bar
reloading plugin Default.sort
reloading plugin Default.swap_line
reloading plugin Default.switch_file
reloading plugin Default.symbol
reloading plugin Default.transform
reloading plugin Default.transpose
reloading plugin Default.trim_trailing_white_space
reloading plugin Default.ui
reloading plugin CSS.css_completions
reloading plugin Diff.diff
reloading plugin HTML.encode_html_entities
reloading plugin HTML.html_completions
reloading plugin 0_package_control_loader.00-package_control
reloading plugin 0_package_control_loader.55-jsonschema
reloading plugin Package Control.1_reloader
reloading plugin Package Control.2_bootstrap
reloading plugin Package Control.Package Control
reloading plugin SCSS.scss_completions
reloading plugin SublimeLinter-contrib-sass-lint.linter
reloading plugin SublimeLinter.__init__
reloading plugin SublimeLinter.active_linters_view
reloading plugin SublimeLinter.busy_indicator_view
reloading plugin SublimeLinter.goto_commands
reloading plugin SublimeLinter.highlight_view
reloading plugin SublimeLinter.log_handler
reloading plugin SublimeLinter.message_view
reloading plugin SublimeLinter.panel_view
reloading plugin SublimeLinter.status_bar_view
reloading plugin SublimeLinter.sublime_linter
plugins loaded
SublimeLinter: SublimeLinter.sublime-package:70: Logging installed; log level INFO
SublimeLinter: SublimeLinter.sublime-package:43: debug mode: on
SublimeLinter: SublimeLinter.sublime-package:44: version: 4.3.3
SublimeLinter: SublimeLinter.sublime-package:229: Delay buffer 57 for 0.3s
SublimeLinter: SublimeLinter.sublime-package:342: detected syntax: scss
SublimeLinter: SublimeLinter.sublime-package:401: Env first path: PATH: /usr/bin:/bin:/usr/sbin:/sbin
SublimeLinter: SublimeLinter.sublime-package:403: Wait until env is loaded but max 60s.
SublimeLinter: SublimeLinter.sublime-package:229: Delay buffer 57 for 0.3s
environment variables loaded using: /bin/bash -l
SublimeLinter: SublimeLinter.sublime-package:387: Env seems ok: PATH: /Users/amclin/.nvm/versions/node/v8.9.3/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
SublimeLinter: #1 SublimeLinter.sublime-package:703: 'sass' is linting '_dynamicmediabanner.scss'
SublimeLinter: #1 SublimeLinter.sublime-package:1113: Running ...

  /Users/amclin/Documents/Clients/VFc/cms/vfcorp-ui/src/design  (working dir)
  $ /Users/amclin/Documents/Clients/VFc/cms/vfcorp-ui/src/design/node_modules/.bin/sass-lint --verbose --no-exit --format stylish /var/folders/z_/s5120h950dxf3794cfsjvd61xb4m48/T/tmpab3yn2

SublimeLinter: #1 SublimeLinter.sublime-package:756: sass output:
    /var/folders/z_/s5120h950dxf3794cfsjvd61xb4m48/T/tmpab3yn2
      1:1  error  Please check validity of the block starting from line #1  Fatal

    ✖ 1 problem (1 error, 0 warnings)
SublimeLinter: #1 SublimeLinter.sublime-package:796: sass: No match for line: ''
SublimeLinter: #1 SublimeLinter.sublime-package:796: sass: No match for line: '/var/folders/z_/s5120h950dxf3794cfsjvd61xb4m48/T/tmpab3yn2'
SublimeLinter: #1 SublimeLinter.sublime-package:796: sass: No match for line: ''
SublimeLinter: #1 SublimeLinter.sublime-package:796: sass: No match for line: '✖ 1 problem (1 error, 0 warnings)'
SublimeLinter: #1 SublimeLinter.sublime-package:796: sass: No match for line: ''
SublimeLinter: SublimeLinter.sublime-package:275: Linting buffer 57 took 0.26s
Package Control: Skipping automatic upgrade, last run at 2018-05-01 17:00:19, next run at 2018-05-01 18:00:19 or after
SublimeLinter: SublimeLinter.sublime-package:229: Delay buffer 58 for 0.3s
SublimeLinter: SublimeLinter.sublime-package:229: Delay buffer 63 for 0.3s
reloading settings Packages/User/Package Control.sublime-settings
```

--- a/linter.py
+++ b/linter.py
@@ -37,7 +37,10 @@ class Sass(NodeLinter):
     regex_error = re.compile(
         r'^(\w*)Error: (?P<msg>.*)'
     )
-    tempfile_suffix = '-'
+    tempfile_suffix = {
+        'scss': 'scss',
+        'sass': 'sass'
+    }
     version_args = '--version'
     version_re = r'(?P<version>\d+\.\d+\.\d+)'
     version_requirement = '>= 1.2.0'
@@ -74,7 +77,7 @@ class Sass(NodeLinter):
 
         return super().find_errors(output)
 
-    def tmpfile(self, cmd, code, suffix=''):
+    def tmpfile(self, cmd, code, suffix=None):
         """Run an external executable using a temp file to pass code and return its output."""
         # check if <style> tag exists
         match_style_open_tag = re.search(r'^\s*<style[^>]*>', code)

